src/coreclr/jit/importervectorization.cpp

```diff
@@ -648,7 +648,24 @@ GenTree* Compiler::impUtf16SpanComparison(StringComparisonKind kind, CORINFO_SIG
             JITDUMP("UTF16 data is too long to unroll - bail out.\n");
             return nullptr;
         }
-        JITDUMP("Trying to unroll MemoryExtensions.Equals|SequenceEqual|StartsWith(op1, \"%ws\")...\n", str)
+
+#ifdef DEBUG
+        if (verbose)
+        {
+            const char* utf8Str = "<utf8 conversion failure>";
+            int         utf8Len = WideCharToMultiByte(CP_UTF8, 0, (WCHAR*)str, -1, nullptr, 0, nullptr, nullptr);
+            if (utf8Len != 0)
+            {
+                char* allocated = new (this, CMK_DebugOnly) char[utf8Len];
+
+                if (WideCharToMultiByte(CP_UTF8, 0, (WCHAR*)str, -1, allocated, utf8Len, nullptr, nullptr) != 0)
+                {
+                    utf8Str = allocated;
+                }
+            }
+            printf("Trying to unroll MemoryExtensions.Equals|SequenceEqual|StartsWith(op1, \"%s\")...\n", utf8Str);
+        }
+#endif
```

Comment:
It looks like too much effort to just dump a string in JitDump, I wouldn't bother

Reply:
Good point - I added a `Compiler::convertUtf16ToUtf8ForPrinting` and switched this to use it.